### ISSUE TITLE
Consumers in melon jobs are not sorted correctly

### ISSUE THREAD

**Issue by tomaszguzialek** [datacleaner/DataCleaner#734]
The job below does not execute properly. The EQUALS filter should filter out "WarszawaPoland" value from Value Distribution, but it is there indeed. It seems, that it happens when there is a transformer and a filter inside the melon scope. The transformer and the filter are not sorted correctly before execution.

![melon_sort_consumer](https://cloud.githubusercontent.com/assets/1483194/10307893/8aecfa30-6c32-11e5-9532-25a8d52c6ef5.png)

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<job xmlns="http://eobjects.org/analyzerbeans/job/1.0">
    <job-metadata>
        <job-description>Created with DataCleaner Community edition UNKNOWN</job-description>
        <author>tomaszg</author>
        <updated-date>2015-10-06+02:00</updated-date>
        <metadata-properties>
            <property name="CoordinatesX.PUBLIC.CUSTOMERS">206</property>
            <property name="CoordinatesY.PUBLIC.CUSTOMERS">78</property>
        </metadata-properties>
    </job-metadata>
    <source>
        <data-context ref="orderdb"/>
        <columns>
            <column id="col_customernumber" path="CUSTOMERS.CUSTOMERNUMBER" type="INTEGER"/>
            <column id="col_customername" path="CUSTOMERS.CUSTOMERNAME" type="VARCHAR"/>
            <column id="col_contactlastname" path="CUSTOMERS.CONTACTLASTNAME" type="VARCHAR"/>
            <column id="col_contactfirstname" path="CUSTOMERS.CONTACTFIRSTNAME" type="VARCHAR"/>
            <column id="col_phone" path="CUSTOMERS.PHONE" type="VARCHAR"/>
            <column id="col_addressline1" path="CUSTOMERS.ADDRESSLINE1" type="VARCHAR"/>
            <column id="col_addressline2" path="CUSTOMERS.ADDRESSLINE2" type="VARCHAR"/>
            <column id="col_city" path="CUSTOMERS.CITY" type="VARCHAR"/>
            <column id="col_state" path="CUSTOMERS.STATE" type="VARCHAR"/>
            <column id="col_postalcode" path="CUSTOMERS.POSTALCODE" type="VARCHAR"/>
            <column id="col_country" path="CUSTOMERS.COUNTRY" type="VARCHAR"/>
            <column id="col_salesrepemployeenumber" path="CUSTOMERS.SALESREPEMPLOYEENUMBER" type="INTEGER"/>
            <column id="col_creditlimit" path="CUSTOMERS.CREDITLIMIT" type="NUMERIC"/>
            <column id="col_jobtitle" path="CUSTOMERS.JOBTITLE" type="VARCHAR"/>
        </columns>
    </source>
    <transformation/>
    <analysis>
        <analyzer>
            <descriptor ref="Completeness analyzer"/>
            <metadata-properties>
                <property name="CoordinatesY">77</property>
                <property name="CoordinatesX">431</property>
            </metadata-properties>
            <properties>
                <property name="Conditions" value="[NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL,NOT_BLANK_OR_NULL]"/>
                <property name="Evaluation mode" value="ANY_FIELD"/>
            </properties>
            <input ref="col_customernumber"/>
            <input ref="col_customername"/>
            <input ref="col_contactlastname"/>
            <input ref="col_contactfirstname"/>
            <input ref="col_phone"/>
            <input ref="col_addressline1"/>
            <input ref="col_addressline2"/>
            <input ref="col_city"/>
            <input ref="col_state"/>
            <input ref="col_postalcode"/>
            <input ref="col_country"/>
            <input ref="col_salesrepemployeenumber"/>
            <input ref="col_creditlimit"/>
            <input ref="col_jobtitle"/>
            <output-data-stream name="Incomplete rows">
                <job>
                    <source>
                        <columns>
                            <column id="col_customernumber2" path="CUSTOMERNUMBER" type="INTEGER"/>
                            <column id="col_customername2" path="CUSTOMERNAME" type="VARCHAR"/>
                            <column id="col_contactlastname2" path="CONTACTLASTNAME" type="VARCHAR"/>
                            <column id="col_contactfirstname2" path="CONTACTFIRSTNAME" type="VARCHAR"/>
                            <column id="col_phone2" path="PHONE" type="VARCHAR"/>
                            <column id="col_addressline12" path="ADDRESSLINE1" type="VARCHAR"/>
                            <column id="col_addressline22" path="ADDRESSLINE2" type="VARCHAR"/>
                            <column id="col_city2" path="CITY" type="VARCHAR"/>
                            <column id="col_state2" path="STATE" type="VARCHAR"/>
                            <column id="col_postalcode2" path="POSTALCODE" type="VARCHAR"/>
                            <column id="col_country2" path="COUNTRY" type="VARCHAR"/>
                            <column id="col_salesrepemployeenumber2" path="SALESREPEMPLOYEENUMBER" type="INTEGER"/>
                            <column id="col_creditlimit2" path="CREDITLIMIT" type="NUMERIC"/>
                            <column id="col_jobtitle2" path="JOBTITLE" type="VARCHAR"/>
                        </columns>
                    </source>
                    <transformation>
                        <transformer>
                            <descriptor ref="Concatenator"/>
                            <metadata-properties>
<property name="CoordinatesY">63</property>
<property name="CoordinatesX">704</property>
                            </metadata-properties>
                            <properties>
<property name="Separator" value="&lt;null&gt;"/>
                            </properties>
                            <input ref="col_city2"/>
                            <input ref="col_country2"/>
                            <output id="col_concatofcitycountry" name="Concat of CITY,COUNTRY"/>
                        </transformer>
                        <filter>
                            <descriptor ref="Equals"/>
                            <metadata-properties>
<property name="CoordinatesY">70</property>
<property name="CoordinatesX">936</property>
                            </metadata-properties>
                            <properties>
<property name="Compare values" value="[WarszawaPoland]"/>
                            </properties>
                            <input ref="col_concatofcitycountry" name="Input column"/>
                            <outcome id="outcome_0" category="INVALID"/>
                        </filter>
                    </transformation>
                    <analysis>
                        <analyzer requires="outcome_0">
                            <descriptor ref="Value distribution"/>
                            <metadata-properties>
<property name="CoordinatesY">66</property>
<property name="CoordinatesX">1363</property>
                            </metadata-properties>
                            <properties>
<property name="Record unique values" value="true"/>
<property name="Record drill-down information" value="true"/>
<property name="Top n most frequent values" value="&lt;null&gt;"/>
<property name="Bottom n most frequent values" value="&lt;null&gt;"/>
                            </properties>
                            <input ref="col_concatofcitycountry" name="Column"/>
                        </analyzer>
                    </analysis>
                </job>
            </output-data-stream>
        </analyzer>
    </analysis>
</job>
```

**Comment by kaspersorensen** [comment]
For anyone that's going to work on this: I Reproduced the issue in branch:

https://github.com/datacleaner/DataCleaner/tree/bug/734-consumer-sorting-in-output-data-stream

in test class ```TransformationAndFilteringInOutputDataStreamsTest```

**Comment by kaspersorensen** [comment]
Oh well, now that it was right in front of me, I couldn't help but fix it :-) See #738
